src/Microsoft.AspNet.OData.Shared/Formatter/ClrTypeCache.cs

```diff
@@ -71,8 +71,13 @@ public bool Equals(EdmTypeCacheItem x, EdmTypeCacheItem y)
 
             public int GetHashCode(EdmTypeCacheItem obj)
             {
-                string combined = $"{obj.EdmType.FullTypeName()}~{obj.Nullable}";
-                return combined.GetHashCode();
+                unchecked
+                {
+                    int hashCode = 17;
+                    hashCode = (hashCode * 31) + obj.EdmType.GetHashCode();
+                    hashCode = (hashCode * 31) + obj.Nullable.GetHashCode();
```

Comment:
@ificator I'm not sure I understand the logic here... What value does line 77 add if the assigned value is overwritten here?

Reply:
argh, never mind. I just realized that the value is not exactly being overwritten

Reply:
Yeah, this is a somewhat standard algorithm for generating hashcodes that's taken from "Effective Java" (https://stackoverflow.com/questions/11537005/why-chose-31-to-do-the-multiplication-in-the-hashcode-implementation). I became less concerned with my change once I found the unit tests for the cache and they were all passing.